### DISCUSSION TITLE
Use LXQt file dialog in firefox

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#2123] [Show and tell]
Just stumpled upon this:
https://www.reddit.com/r/kde/comments/a5cxwk/firefox_v64_can_now_use_the_kde_file_selection/
Starting firefox with `GTK_USE_PORTAL=1 /usr/bin/firefox %u`  tricks it into using the LXQt filedialog.
https://github.com/lxqt/xdg-desktop-portal-lxqt has to be installed.

EDIT: works also for thunderbird and probably other GTK apps.

**Comment by redtide** [comment]
In this specific case it can be set permanently via `about:config` by setting `widget.use-xdg-desktop-portal` to `true`.

**Reply by stefonarch** [reply]
Doesn't work for me in both...

**Reply by redtide** [reply]
sorry, both what? I'm using Arch with the LXQt portal installed, so it should work also for you

**Reply by stefonarch** [reply]
On arch here atm nothing works with those settings either as prefix or in about:config : no filedialog at all. No idea what changed.

**Reply by redtide** [reply]
I had some troubles too in the beginning, but [they helped me to solve it](https://github.com/lxqt/lxqt/issues/2069#issuecomment-967206609), though now I'm not able to [uninstall it](https://github.com/lxqt/lxqt/issues/2069#issuecomment-1006542411) probably due to the reference cycle. To me it's fine for now, I don't need to remove it but still an annoying thing.

**Reply by stefonarch** [reply]
This has to be continued in the issue probably. As I started that discussion I was on debian, now I restarted debian and the change in about:config is immediate and works for and back. On both machines using arch  not.
```
ls /usr/share/xdg-desktop-portal/portals/
gnome-keyring.portal  lxqt.portal
```

**Reply by stefonarch** [reply]
Found one way: 
* install xdg-desktop-portal, which gives 4 providers  on arch (gnome, gtk, kde, wlr)
* select 3) kde
* remove manually  /usr/share/xdg-desktop-portal/portals/kde.portal
* reboot

**Reply by redtide** [reply]
That's the way I used in the "Chrome 93" issue to install it. I think you should install and uninstall xdg-desktop-portal-kde, not removing it manually. It's a workaround because the pseudo impl and the recursive dependency in xdg-desktop-portal-lxqt package I think.
I was waiting for some feedback from LXQt people but at this point you could open an issue, but not only for the installation, it's more for uninstalling the lxqt portal, the issue you are facing is the same I had and they know already about it.

**Reply by stefonarch** [reply]
One cannot uninstall only xdg-desktop-portal-kde:
```
 yay -R xdg-desktop-portal-kde
checking dependencies...
error: failed to prepare transaction (could not satisfy dependencies)
:: removing xdg-desktop-portal-kde breaks dependency 'xdg-desktop-portal-impl' required by xdg-desktop-portal

```
It's more an issue of this *-impl in arch depencies, not our problem. Once released xdg-desktop-portal-lxqt it will be probably added to the providers.

**Reply by redtide** [reply]
just tried and I was right, you need to install any other xdg-desktop-portal-impl (where impl is one of kde, gtk, etc) to be able to install the lxqt version, the same for uninstall, you install any of the above, uninstall the lxqt one, then uninstall the rest using `pacman -Rs xdg-desktop-portal`. That impl thing seems tricky, yes. If needed I'll tell you the steps I did later, I need to update the system first.

**Reply by redtide** [reply]
now I have other problems here to install again the lxqt impl, can you please try to remove all the portal packages as I said above and retry to reinstall the lxqt one? I need you to confirm my current issue.
If for some reason you can't uninstall the kde one, try to install again on top of it that you manually removed first, then remove all packages with `pacman -Rs xdg-desktop-portal`, in which case let me know if you did this step and worked.
Please don't remove files manually or you risk to compromise the system.

**Reply by redtide** [reply]
Meanwhile I opened [an issue](https://github.com/lxqt/xdg-desktop-portal-lxqt/issues/1).
